src/app/qgsgeometryvalidationdock.cpp

```diff
@@ -159,6 +161,35 @@ void QgsGeometryValidationDock::onRowsInserted()
   setUserVisible( true );
 }
 
+void QgsGeometryValidationDock::showErrorContextMenu( const QPoint &pos )
+{
+  bool showUnreliableResolutionMethods = QgsSettings().value( QStringLiteral( "geometry_validation/enable_problem_resolution" ) ) == QLatin1String( "true" );
+
+  QModelIndex index = mErrorListView->indexAt( pos );
+  QgsGeometryCheckError *error = index.data( QgsGeometryValidationModel::GeometryCheckErrorRole ).value<QgsGeometryCheckError *>();
+  if ( error )
+  {
+    const QList<QgsGeometryCheckResolutionMethod> resolutionMethods = error->check()->availableResolutionMethods();
+
+    QMenu *menu = new QMenu( this );
```

Comment:
This is a quasi-leak, because the menus & actions will build up in memory over time and won't be deleted until the dock is (i.e. the lifetime of the qgis session). The approach used elsewhere is to make the menu a member of the widget, and then clear it here before populating it. That way there's only ever once instance of the menu in memory.